formats/json/commonMain/src/kotlinx/serialization/json/internal/JsonElementMarker.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+@file:Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")
+
+package kotlinx.serialization.json.internal
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.internal.ElementMarker
+
+@OptIn(ExperimentalSerializationApi::class)
+internal class JsonElementMarker(descriptor: SerialDescriptor) {
+    private val origin: ElementMarker = ElementMarker(descriptor, ::readElementIfAbsence)
+
+    internal var isUnmarkedNull: Boolean = false
+        private set
+
+    internal fun mark(index: Int) {
+        origin.mark(index)
+    }
+
+    internal fun nextUnmarkedIndex(): Int {
+        return origin.nextUnmarkedIndex()
+    }
+
+    private fun readElementIfAbsence(descriptor: SerialDescriptor, index: Int): Boolean {
```

Comment:
`ifAbsence` -> `ifAbsent`.
Also, `element` is probably redundant, i.e. `readIfAbsent` seems concise enough